056-kafka-roller-kraft.md

```diff
@@ -0,0 +1,129 @@
+# Kafka Roller KRaft Support
+
+This proposal describes the actions that the KafkaRoller should take when operating 
+against a Strimzi cluster in KRaft mode.
+The proposal omits details about exactly how this should be implemented because 
+the logic of the KafkaRoller is currently being rewritten.
+This proposal only describes the expected behaviour so that it can be implemented 
+in either the current KafkaRoller or a future iteration of the KafkaRoller.
+
+This proposal assumes that liveness/readiness of nodes is as described in [proposal #46](https://github.com/strimzi/proposals/blob/main/046-kraft-liveness-readiness.md) 
+and [PR #8892](https://github.com/strimzi/strimzi-kafka-operator/pull/8892).
+
+
+## Current situation
+
+When operating on a ZooKeeper based cluster the KafkaRoller has the behaviour 
+described below.
+
+### Order to roll nodes
+Roll in order:
+ - Unready brokers
+ - Ready brokers
+ - Current controller
+
+### Triggers
+The following triggers an attempt to roll a broker:
+- Pod is annotated for manual rolling update
+- Pod is unready
+- Broker's configuration has changed and cannot be updated dynamically
+
+### Rollability checks
+ - Attempts to connect an admin client to the broker and if it can't connect rolls the broker
+ - Does not force roll a broker performing log recovery
+ - Does not roll a broker if doing so would take the in-sync replicas count below `min.insync.replicas`
+
+### Configuration changes
+ - Compares the current configuration with expected configuration to see if a broker needs rolling
```

Comment:
I don't think it is really clear what configuration means here. The Kafka configuration itself? This might deserve more detail about how it works as it will highlight the differences.